src/Tizen.Nlp/Tizen.Nlp/NlpHandle.cs

```diff
@@ -0,0 +1,149 @@
+/*
+* Copyright (c) 2018 Samsung Electronics Co., Ltd All Rights Reserved
+*
+* Licensed under the Apache License, Version 2.0 (the License);
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an AS IS BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using Tizen.Applications;
+
+namespace Tizen.Nlp
+{
+
+    /// <summary>
+    /// This class contains the methods and inner class related to the NLP processing.
+    /// </summary>
+    /// <since_tizen> 5 </since_tizen>
+    public class NlpHandle
+    {
+        /// <summary>
+        /// An EventHandler to expose to external to recieve data from remote Nlp service  .
+        /// </summary>
+        public event EventHandler<MessageReceivedEventArgs> MessageReceived;
+        private Message _msg;
+        private readonly Message.NotifyCb _noti = new Message.NotifyCb();
+        private string _tag;
+
+        private void MakeRequest(string cmd, string info)
+        {
+            Bundle b = new Bundle();
+            b.AddItem("command", cmd);
+            b.AddItem("info", info);
+            _msg.Send(b);
+        }
+
+        private void OnReceived(string sender, Bundle msg)
+        {
+            Log.Debug(_tag, "OnReceived ++");
+            MessageReceivedEventArgs e = new MessageReceivedEventArgs();
+            e.Message = msg;
+            MessageReceived?.Invoke(this, e);
+            Log.Debug(_tag, "done");
+        }
+
+        /// <summary>
+        /// An init session to connect remote tidl service with 2 input parameters.
+        /// </summary>
+        /// <param name="serviceId">remote nlp service id.</param>
+        /// <param name="clientId">local nlp client app id.</param>
+        /// <returns></returns>
+        /// <since_tizen> 5 </since_tizen>
+        public void Init(string serviceId, string clientId)
```

Comment:
Why do you need 2 phase creation? (constructor -> init)
How about use Constructor to initialize?